src/main/java/org/jboss/xavier/integrations/route/VMWorkloadInventoryRoutes.java

```diff
@@ -28,8 +30,23 @@ public void configure() throws Exception {
         from("direct:calculate-vmworkloadinventory").routeId("calculate-vmworkloadinventory")
             .setHeader("KieSessionId", constant("WorkloadInventoryKSession0"))
             .bean("VMWorkloadInventoryCalculator", "calculate(${body}, ${header.${type:org.jboss.xavier.integrations.route.MainRouteBuilder.MA_METADATA}})", false)
-            .split(body()).parallelProcessing(parallel).aggregationStrategy(new WorkloadInventoryReportModelAggregationStrategy())
+                .process(exchange -> {
+                    Set<String> vmNamesWithSharedDisk = exchange.getIn().getHeader("vmNamesWithSharedDisk", Set.class);
+                    Collection<VMWorkloadInventoryModel>  allVms = exchange.getIn().getBody(Collection.class);
+                    Collection<VMWorkloadInventoryModel> originalAndAmendedVms = allVms.stream().map(vmWorkloadInventoryModel ->
+                    {
+                        if (vmNamesWithSharedDisk != null && vmNamesWithSharedDisk.contains(vmWorkloadInventoryModel.getVmName())) {
+                            vmWorkloadInventoryModel.setHasSharedVmdk(true);
+                        }
+                        return vmWorkloadInventoryModel;
+                    }).collect(Collectors.toList());
+                    exchange.getIn().setBody(allVms);
+                    exchange.getIn().removeHeader("vmNamesWithSharedDisk");
+                    exchange.getIn().removeHeader("originalBody");
```

Comment:
I don't see this block of code is adding value while analyzing it against xavier-analytics. Please see my comment below for more info.